### PR TITLE
Manually trigger compaction restapi interface

### PR DESCRIPTION
**Pull request by ZhangYu0123**
## Proposed changes

Add restapi to be which do compaction task by manual trigger.  The detail  design in #4311 .

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on #ISSUE, and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

File: be/src/http/action/compaction_action.cpp [hunk 1]
```diff
@@ -67,6 +72,127 @@ Status CompactionAction::_handle_show_compaction(HttpRequest* req, std::string*
...
+        if (olap_status != OLAP_SUCCESS) {
+            return Status::InternalError(
+                    strings::Substitute("fail to execute compaction, error = $0", olap_status));
+        }
+    } else {
+        LOG(INFO) << "Manual compaction task is timeout for waiting " << (status == std::future_status::timeout);
+    }
+   
+    LOG(INFO) << "Manual compaction task is successfully trigged";
+    *json_result = "{\"status\": \"Success\", \"msg\": \"compaction task is successfully trigged.\"}";
```
Comment (morningman) [inline be/src/http/action/compaction_action.cpp:147]: trigged -> triggered

File: be/src/http/action/compaction_action.h [hunk 2]
```diff
@@ -41,8 +52,25 @@ class CompactionAction : public HttpHandler {
...
+    /// fetch compaction running status
+    Status _handle_run_status_compaction(HttpRequest* req, std::string* json_result);
+
 private:
     CompactionActionType _type;
+
+    /// running check mutex
+    static std::mutex _compaction_running_mutex;
+    /// whether there is manual compaction running
+    static bool _is_compaction_running;
```
Comment (acelyc111) [inline be/src/http/action/compaction_action.h:71]: Is an atomic\<bool> OK?

Comment (ZhangYu0123) [reply be/src/http/action/compaction_action.h:74]: already use mutex

File: be/src/http/action/compaction_action.cpp [hunk 3]
```diff
@@ -30,33 +31,52 @@
...
+// for viewing the compaction status
+Status CompactionAction::_handle_show_compaction(HttpRequest* req, std::string* json_result) {
+    
+    uint64_t tablet_id = 0;
+    uint32_t schema_hash = 0;
+    
+    Status status = _check_param(req, &tablet_id, &schema_hash);
+    if (!status.ok()) {
+        return status;
+    }
```
Comment (acelyc111) [inline be/src/http/action/compaction_action.cpp:78]: You can use macro `RETURN_IF_ERROR` to simplify code.

File: be/src/http/action/compaction_action.h (hunk 2 above)
Comment (acelyc111) [reply be/src/http/action/compaction_action.h:74]: I mean, mutex is not needed, you can use atomic<bool> instead, I found `_compaction_running_mutex ` is only aim to protect `_is_compaction_running `

File: docs/en/administrator-guide/http-actions/compaction-action.md [hunk 4]
```diff
@@ -95,4 +95,96 @@ curl -X GET http://192.168.10.24:8040/api/compaction/show?tablet_id=10015\&schem
 
 ## Manually trigger Compaction
 
-(TODO)
+```
+curl -X GET http://be_host:webserver_port/api/compaction/run?tablet_id=xxxx\&schema_hash=yyyy\&compact_type=cumulative
```
Comment (imay) [inline docs/en/administrator-guide/http-actions/compaction-action.md:99]: Why use GET method? I think POST method is more suitable.

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/http/action/compaction_action.h (hunk 2 above)
Comment (ZhangYu0123) [reply be/src/http/action/compaction_action.h:74]: Only use atomic<bool> can not cover this scenes, becouse atomic<bool> do not have test and set operation.

File: docs/en/administrator-guide/http-actions/compaction-action.md [hunk 5]
```diff
@@ -95,4 +95,96 @@ curl -X GET http://192.168.10.24:8040/api/compaction/show?tablet_id=10015\&schem
...
+### Examples
+
+```
+curl -X POST http://192.168.10.24:8040/api/compaction/run?tablet_id=10015\&schema_hash=1294206575\&compact_type=cumulative
+```
+
+## Manual Compaction execution status
+
+```
+curl -X GET http://be_host:webserver_port/api/compaction/run?tablet_id=xxxx\&schema_hash=yyyy
```
Comment (imay) [inline docs/en/administrator-guide/http-actions/compaction-action.md:145]:
```suggestion
curl -X GET http://be_host:webserver_port/api/compaction/run_status?tablet_id=xxxx\&schema_hash=yyyy
```

File: docs/en/administrator-guide/http-actions/compaction-action.md (hunk 4 above)
Comment (ZhangYu0123) [reply docs/en/administrator-guide/http-actions/compaction-action.md]: The document is wrong and has been modified

File: be/src/http/action/compaction_action.cpp [hunk 6]
```diff
@@ -67,6 +82,167 @@ Status CompactionAction::_handle_show_compaction(HttpRequest* req, std::string*
...
+        "schema_hash" : $3,
+        "compact_type" : "$4"
+})";
+
+    std::string msg = "this tablet_id is not running";
+    std::string compaction_type = "";
+    bool run_status = 0;
+
+    // use try lock to check this tablet is running cumulative compaction
+    MutexLock lock_cumulativie(tablet->get_cumulative_lock(), TRY_LOCK);
```
Comment (morningman) [inline be/src/http/action/compaction_action.cpp:183]: Miss to unlock?

**Review by morningman — APPROVED** [review]
LGTM

**Review by imay — APPROVED** [review]

**Pull request merged**